pages/profile/index.js

```diff
@@ -156,19 +193,58 @@ export default function ProfilePage({
           toggleMenu={() => setIsNotificationVisible(false)}
         >
           {user.connectionRequests.length >= 1 ? (
-            user.connectionRequests.map((request, index) => (
-              <div key={request.senderId}>
-                <p>{request.message}</p>
-                <div>
-                  <button onClick={() => addFriend(request.senderId, index)}>
-                    bestätigen
-                  </button>
-                  <button onClick={() => rejectFriendRequest(index)}>
-                    ablehnen
-                  </button>
-                </div>
-              </div>
-            ))
+            user.connectionRequests.map((request, index) => {
+              if (request.type === 1)
+                return (
+                  <div key={request.senderId}>
+                    <p>{request.message}</p>
+                    <div>
+                      <button
+                        onClick={() =>
+                          acceptFriendRequest(request.senderId, index)
+                        }
+                      >
+                        bestätigen
+                      </button>
+                      <button
+                        onClick={() =>
+                          rejectFriendRequest(request.senderId, index)
+                        }
+                      >
+                        ablehnen
+                      </button>
+                    </div>
+                  </div>
+                );
+              if (request.type === 3)
+                return (
+                  <div key={request.senderId}>
+                    <p>{request.message}</p>
+                    <div>
+                      <button
+                        onClick={() =>
+                          acceptNewHousehold(
+                            request.senderId,
+                            request.householdId
+                          )
+                        }
+                      >
+                        bestätigen
+                      </button>
+                      <button
+                        onClick={() =>
+                          rejectNewHousehold(
+                            request.senderId,
+                            request.householdId
+                          )
+                        }
+                      >
+                        ablehnen
+                      </button>
+                    </div>
+                  </div>
+                );
+            })
```

Comment:
If it is just the function call that changes here, we could return one element, and then in the onclick do a check for the value of request.type.